src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/FakeRoleAssignmentsGenerator.java

```diff
@@ -0,0 +1,108 @@
+                                        @Qualifier(CachedCaseDefinitionRepository.QUALIFIER)
+                                            CaseDefinitionRepository caseDefinitionRepository) {
+        this.userRepository = userRepository;
+        this.caseDefinitionRepository = caseDefinitionRepository;
+    }
+
+    public List<RoleAssignment> addFakeRoleAssignments(List<RoleAssignment> roleAssignments) {
+        List<String> idamUserRoles = new ArrayList<>(userRepository.getUserRoles());
+        List<String> grantedRoles = filterGrantedRoles(idamUserRoles);
+        List<RoleAssignment> augmentedRoleAssignments;
+
+        if (!grantedRoles.isEmpty()) {
+            if (atLeastOneCaseRoleExists(roleAssignments)) {
+                augmentedRoleAssignments = concat(roleAssignments.stream(),
+                                                  createFakeRoleAssignmentsGrantSpecific(grantedRoles).stream())
+                    .collect(Collectors.toList());
+            } else {
+                augmentedRoleAssignments = new ArrayList<>(roleAssignments);
+            }
+        } else {
+            augmentedRoleAssignments = concat(roleAssignments.stream(),
+                                              createFakeRoleAssignmentsGrantStandard(idamUserRoles).stream())
+                .collect(Collectors.toList());
+        }
+
+        return augmentedRoleAssignments;
+    }
+
+    private List<RoleAssignment> createFakeRoleAssignmentsGrantSpecific(List<String> grantedRoles) {
+        return grantedRoles.stream()
+            .map(role -> RoleAssignment.builder()
+                .roleName(IDAM_PREFIX + role)
+                .grantType(GRANT_TYPE_SPECIFIC)
+                .classification(CLASSIFICATION_RESTRICTED)
+                .build())
+            .collect(Collectors.toList());
+    }
+
+    private List<RoleAssignment> createFakeRoleAssignmentsGrantStandard(List<String> grantedRoles) {
+        List<UserRole> classifications = caseDefinitionRepository
+            .getClassificationsForUserRoleList(grantedRoles);
+
+        Map<String, String> roleToClassification = classifications.stream()
+            .collect(Collectors.toMap(UserRole::getRole, UserRole::getSecurityClassification));
+
+        return grantedRoles.stream()
+            .map(role -> RoleAssignment.builder()
+                .roleName(IDAM_PREFIX + role)
+                .grantType(GRANT_TYPE_STANDARD)
+                .classification(roleToClassification.get(role))
+                .build())
+            .collect(Collectors.toList());
+    }
+
+    private boolean atLeastOneCaseRoleExists(List<RoleAssignment> roleAssignments) {
+        return roleAssignments.stream()
+            .anyMatch(this::isCaseRole);
+    }
+
+    private boolean isCaseRole(RoleAssignment roleAssignment) {
+        return roleAssignment.getRoleType().equalsIgnoreCase(ROLE_TYPE_CASE);
```

Comment:
Changes to: "IF there is at least one remaining role assignment containing a case Id attribute after filtering". We have to check the caseId on the RoleAssignmentAttributes.
Missing caseId -> Optional.empty
caseId: null -> Optional.of("null")
caseId: 12345 -> Optional.of(12345")